analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -122,19 +126,33 @@ internal fun SpdxExternalDocumentReference.getSpdxPackage(packageId: String, wor
 /**
  * Return the [SpdxDocument] this [SpdxExternalDocumentReference]'s [SpdxDocument] refers to.
  */
-private fun SpdxExternalDocumentReference.resolve(workingDir: File): SpdxDocument {
+private fun SpdxExternalDocumentReference.resolve(workingDir: File, issues: MutableList<OrtIssue>): SpdxDocument {
```

Comment:
A design question: would it not be cleaner to return `Pair<SpdxDocument, MutableList<OrtIssue>` or a data class, than to modify a parameter ?

Reply:
Personally, I actually like the semantics of "out" parameters; while it would be nice if Kotlin had a dedicated `out` keyword like C#, I believe it's sufficiently documented via the `*Mutable*List`. And last but not least, using out-parameters has better performance as it avoids the temporary creation of collections (and the their merge).

Reply:
I understand your reasoning and I think this is a taste thing again. Personally, I find `out` parameters a code smell as it requires mutable classes. I like how destructuring declaration in Kotlin addresses the overhead e.g. `val (document, issues) = reference.resolve` .

Since this is a taste thing and ORT already follow this coding style, this is not a blocker for me.